utils/external_service/base_external_service_client.py

```diff
@@ -79,6 +80,7 @@ def handle_500_error(cls, response: Response) -> None:
     ################
 
     @classmethod
+    @stamina.retry(on=Exception, attempts=3)  # Likely retry should happen no matter the exception
```

Comment:
Or exclude NotFound, Conflict, BadRequest?

Reply:
I think the way this is configured now, the retry only occurs if there is an issue in getting a response (ConnectionError, Timout, etc.), and not on any valid response from the server (400, 404, 409). See possible exceptions [here](https://github.com/psf/requests/blob/main/src/requests/exceptions.py). Not sure if this covers 500 errors, I should check.

Reply:
Checked and no, this does not retry on 500 errors. I'll fix the implementation.

Reply:
Retry logic should be fixed now, added tests.